src/agent/agent_transport.rs

```diff
@@ -27,7 +27,7 @@ impl Agent {
             // block until pair selected
             tokio::select! {
                 _ = on_connected_rx.recv() => {},
-                _ = cancel_rx.recv() => {
+                Some(()) = cancel_rx.recv() => {
```

Comment:
I don't see any justification for this change.

With this change, drop(cancel_tx) won't cancel the dail and accept, which is not acceptable.